torchvision/ops/poolers.py

```diff
@@ -224,7 +224,7 @@ def forward(self, x, boxes, image_shapes):
             if torchvision._is_tracing():
                 tracing_results.append(result_idx_in_level.to(dtype))
             else:
-                result[idx_in_level] = result_idx_in_level
+                result[idx_in_level] = result_idx_in_level.to(result.dtype)
```

Comment:
instead of casting the result back to fp16 in the user code, I wonder if we shouldn't cast back to `fp16` in the `ROIAlign_autocast` function, after performing the function call?

Reply:
Can you see if removing this now works?

Reply:
Without the Python-side cast, i was getting type mismatch errors thrown from an `index_put` (or maybe `index_put_`?  `index_put` calls `index_put_` so it could have been either).  It inspired
https://github.com/pytorch/pytorch/pull/41035
https://github.com/pytorch/pytorch/pull/41036

`result[idx_in_level] = result_idx_in_level...` is equivalent to an in-place `index_put_`, so
- https://github.com/pytorch/pytorch/pull/41036 doesn't help.  autocast only works on out-of-place ops.
- Even though `ROIAlign_autocast` now returns a tensor matching the type of `roi_align`'s input (in this case `per_level_feature`) it isn't obvious to me that `result` and `result_idx_in_level`'s dtypes will always match.  In particular, if tensors in `x_filtered` have different dtypes, I think `result.dtype != result_idx_in_level.dtype` for some iterations.  I'd rather keep the cast.  It also makes the branch more symmetric with the `_is_tracing()` branch.

Reply:
Fair points. Can you make a comment that this casting is for autocast?